### DISCUSSION TITLE
Unable to fetch data from the parent loader using `useRouteDataLoader`

### DISCUSSION THREAD

**Discussion by devraj** [remix-run/remix#7546] [Q&A] [answered]
My routes are structured like this:

```sh
app/routes/teacher.$classId._index.tsx
app/routes/teacher.$classId.learn._index.tsx
app/routes/teacher.$classId.learn.video.$videoId.tsx
```

Where I send a user to `/teacher/{id}` where they see a menu, they then follow their way to ``/teacher/{id}/learn` where they are presented with a list of videos they can watch and then finally `/teacher/{id}/learn/video/{id}`

When I check the paths using `useMatches` I can see `/teacher` and `/teacher/{id}/learn/video/{id}`

```tsx
    const matches = useMatches();
    console.log("matches", matches);
```
<img width="1163" alt="Screenshot 2023-09-27 at 1 17 02 pm" src="https://github.com/remix-run/remix/assets/5961/0f5e07ee-5cbc-4590-a565-4a04d7a5e98e">

If I try and access the `loader` of the route above it (in my case to get a list of videos) I get an `undefined`.

```tsx
    const data = useRouteLoaderData(`routes/teacher.$classId.learn`);
    console.log("route data", data);
```

If fetch the `loader` of `/teacher` I am able to get the data from the loader.

Is it not possible to get access the loader of a path that is not accessible by `useMatches`. Or am I not understanding `useRouteLoaderData` properly?

Any pointers will be greatly appreciated. Thanks for your time.

**Comment by sergiodxa** [comment, marked as answer]
The useRouteLoaderData doesn’t fetch anything, it’s just a convenient way to access the loader data of a route that it’s currently matching, similar to using `useMatches().find(route => route.id === routeId)?.data`.

If you want to fetch the data of another loader use the useFetcher hook, but this will happen client side, so I recommend you to just return from the route loader the data you need.

**Reply by devraj** [reply]
I ended up doing that anyway, good to know that returning from the route's loader is the correct way to approach it. Thank you for taking the time to respond to me.

**Reply by kiliman** [reply]
I see this confusion often. A lot of experienced React developers are so used to fetching from the client, that when they see hooks like `useLoaderData()`, they automatically assume this is how you *fetch* the data, instead of accessing the *already* fetched data from your loader.

I think it would be helpful if they called this out more in the documentation. Something like:

>This hook does not *fetch* the data but returns the data already returned from your loader and stored in the Remix context. If you need to fetch data from your client, then use the `useFetcher` hook.
